cpp/src/groupby/groupby.cu

```diff
@@ -74,16 +74,8 @@ std::pair<std::unique_ptr<table>, std::vector<aggregation_result>> groupby::disp
   // Only use hash groupby if the keys aren't sorted and all requests can be
   // satisfied with a hash implementation
   if (_keys_are_sorted == sorted::NO and not _helper and
-      detail::hash::can_use_hash_groupby(_keys, requests)) {
-    // Optionally flatten nested key columns.
-    auto flattened             = flatten_nested_columns(_keys, {}, {}, column_nullability::FORCE);
-    auto flattened_keys        = flattened.flattened_columns();
-    auto is_supported_key_type = [](auto col) { return cudf::is_equality_comparable(col.type()); };
-    CUDF_EXPECTS(std::all_of(flattened_keys.begin(), flattened_keys.end(), is_supported_key_type),
-                 "Unsupported groupby key type does not support equality comparison");
-    auto [grouped_keys, results] =
-      detail::hash::groupby(flattened_keys, requests, _include_null_keys, stream, mr);
-    return std::pair(unflatten_nested_columns(std::move(grouped_keys), _keys), std::move(results));
```

Comment:
`unflatten_nested_columns` seems to be used only here. So I would recommend removing it too.

Reply:
Where do we have the leftover of `unflatten_nested_columns`?

Reply:
It should be in `structs/utilities.hpp|cpp`.

Reply:
Ah, I see what you mean then. I will create another PR to remove it once the current work gets merged

Reply:
Filed #10952 to track this issue separately.